docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,190 @@
+
+=== Overriding settings
+
+Once you have built a server image using the `build` command, you should be able to override any property that is available from the `start` command.
+
+For instance, if you have built an image as follows:
+
+<@kc.build parameters="--db=postgres --db-url-host=keycloak-postgres --db-username=keycloak --db-password=change_me --hostname mykeycloak.acme.com"/>
+
+You should be able to override the `db-username`, `hostname`, or any other property available to the `start` command:
+
+<@kc.start parameters="--db-username foo --hostname newkeycloak.acme.com"/>
+
+As a result, the server should start using all the configuration options previously set using the `build` command but overriding those that you explicitly
+set when starting the server.
+
+This configuration approach can be useful when building an opinionated server image with the possibility to override specific options that are allowed to
+change when starting the server (those available to the `start` command).
+
+== Using placeholders
+
+You should be able to use placeholders to dynamically change the value of configuration options when starting the server.
+
+For that, you should use the following format when defining the value of an option:
+
+```
+db.username=${r"${my.db.user}"}
+```
+
+By using a placeholder like in the example above, the server is going to try to resolve the `my.db.user` variable from:
+
+* A system property using the same key
+* An environment variable by replacing dots with `_`.
+
+Placeholders can also be used when building a server image using the `build` command so that the value of a configuration option is updated
+with the value set to the placeholder (system property or environment variable) when starting the server.
+
+[NOTE]
+You should not use placeholders for configuration options that can only be set through the `build` command. The reason is that these options
+can not change once they are persisted into the server image. Prefer using placeholders only for the configurations options available to
+the `start` command.
+
+== Configuration profiles
+
+Most of the time the server is deployed to different environments like testing, staging, and production environments. Although they share some
+core configuration options (e.g: database, clustering, etc) some other options might change on a per-environment basis.
+In order to make easier to support multiple configurations, the server allows you to create configuration profiles.
+
+Configuration profiles are basically a set of one or more configuration options grouped together using a unique key.
+
+You can activate a profile by using the `[-pf|--profile]` option as follows:
+
+<@kc.all rootParameters="-pf myprofile"/>
+
+Once a profile is activated, all the configuration options defined for the profile should take effect when configuring or starting the server.
+
+=== Understanding the default profiles
+
+By default, the server defines two main profiles:
+
+* `dev`
```

Comment:
shall we mention the 'test' profile which should not be used bc. it's internal?

Reply:
I don't think so. Just because we don't have a default `test` profile.

But that is a good point. TBH, I do not know the impact of activating the `test` profile due to the changes we did to our test suite. I mean, people should be able to use `test` as a key.

Reply:
Yes, in general it works, see attached screenshot, and I'm also not sure if we should discourage ppl to use test as key.
![Screenshot 2022-01-11 at 20 13 18](https://user-images.githubusercontent.com/89905860/149006630-94d53182-48d5-48a6-b5b4-9c99bd248f37.png)

Reply:
We should not.